pkg/util/cluster/cluster.go

```diff
@@ -356,22 +357,103 @@ func (c *Cluster) Create(ctx context.Context, vnetResourceGroup, clusterName str
 	return nil
 }
 
+// ipRangesContainCIDR checks, weather any of the ipRanges overlap with the cidr string. In case cidr isn't valid, false is returned.
+func ipRangesContainCIDR(ipRanges []*net.IPNet, cidr string) bool {
+	_, cidrNet, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return false
+	}
```

Comment:
I am not sure if this is a good idea as in _generateSubnets()_ we actually return a "good" result when _ipRangesContainCIDR_ returns false, so maybe this could bring wrong values when returning "false" from _generateSubnets()_ due to an error and not due to a real non overlapping condition. I think it would be great to return an error from this function (if any)

Something like:
```go
// ipRangesContainCIDR checks, weather any of the ipRanges overlap with the cidr string
// and an error, if any.
func ipRangesContainCIDR(ipRanges []*net.IPNet, cidr string) (bool, error) {
	_, cidrNet, err := net.ParseCIDR(cidr)
	if err != nil {
		return false, err
	}

	for _, snet := range ipRanges {
		if snet.Contains(cidrNet.IP) || cidrNet.Contains(snet.IP) {
			return true, nil
		}
	}
	return false, nil
}
```
 and then in _generateSubnets_ change the if condition (check next comment).